File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/features/auth/changePassword/ChangePasswordRequest.java
package br.com.deltaglobalbank.identity.features.auth.changePassword;

import jakarta.validation.constraints.NotBlank;

public record ChangePasswordRequest(
    @NotBlank(message = "currentPassword é obrigatório")
    String currentPassword,

    @NotBlank(message = "newPassword é obrigatório")
    String newPassword
) {
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/features/auth/changePassword/ChangePasswordRequest.kt
package br.com.deltaglobalbank.identity.features.auth.changePassword

import jakarta.validation.constraints.NotBlank

data class ChangePasswordRequest(
    @field:NotBlank(message = "currentPassword é obrigatório")
    val currentPassword: String,

    @field:NotBlank(message = "newPassword é obrigatório")
    val newPassword: String
)